### PR TITLE
Format titles on single issue / PR pages

### LINKED ISSUE #1984: Format titles on single issue / PR pages
[Issuehunt badges](https://issuehunt.io/r/sindresorhus/refined-github/issues/1984)

image

This should be formatted the same as the above:

image

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Sponsors (Total: $40.00)

- [issuehunt issuehunt](https://issuehunt.io/u/issuehunt) ($40.00)

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/refined-github/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
</details>

### PR DESCRIPTION
**Pull request by HardikModha**
Closes #1984

### Test
1. For issues: All issues present on https://github.com/HardikModha/github-playground/issues

2. For PR pages: All PRs present on https://github.com/HardikModha/github-playground/pulls

3. Also tested the feature by updating the issue / PR title. (https://github.com/HardikModha/github-playground/issues/9)

### CODE REVIEW

File: source/features/linkify-issues-in-titles.tsx [hunk 1]
```diff
@@ -1,17 +1,26 @@
...
 
 function init(): void {
 	observeEl(
 		select('#partial-discussion-header')!.parentElement!,
 		() => {
-			const title = select('.js-issue-title:not(.refined-linkified-title)');
-			if (title) {
+			const title = select('.js-issue-title')!;
+
+			if (!title.classList.contains('refined-linkified-title')) {
```
Comment (bfred-it) [inline source/features/linkify-issues-in-titles.tsx]: This change needs to be undone; the class ensures that what follows it is not repeated over and over after ajax updates. _"What follows it"_ now includes parsing the backticks.

**Comment by HardikModha** [comment]
Not related to this, But when I was working on this feature I noticed this, 

Screen Shot 2019-06-02 at 3 32 22 PM

It is **not** misaligned but it looks like it is because the text is always kept on right. Can we keep the text to left aligned?

**Comment by HardikModha** [comment]
I couldn't find it. But seems like it was added https://github.com/sindresorhus/refined-github/commit/767791ec1a1ba6f6c150a4516d120988593b39af#diff-0730bb7c2e8f9ea2438b52e419dd86c9R13 but later removed accidentally in https://github.com/sindresorhus/refined-github/commit/3298fc72de3e0feed35ed4e0d43699ac976e1e11#diff-0730bb7c2e8f9ea2438b52e419dd86c9L35

**Comment by bfred-it** [comment]
Ah, indeed. It's now part of the Highlights:

> Clickable references to issues and pull requests

**Comment by bfred-it** [comment]
The screenshot can stay in the feature but the readme already mentions this feature in two parts, so I dropped it from there:

> - Clickable references to issues and pull requests
> - Text wrapped in backticks in issue titles and commit titles is highlighted

**Comment by bfred-it** [comment]
Thanks for the PR!

At some point I'd probably want to see a full `formatText()` function that can be used across the extension (here, `parse-backticks`, `options.tsx`, etc)

**Comment by HardikModha** [comment]
Sorry but didn't get what you meant by this. `parseBackticks` is already being used in `options.tsx`. What this `formatText` function do additionally? Will it take care of backticks as well as links?

**Comment by HardikModha** [comment]
> Not related to this, But when I was working on this feature I noticed this,
> 
> Screen Shot 2019-06-02 at 3 32 22 PM
> 
> It is **not** misaligned but it looks like it is because the text is always kept on right. Can we keep the text left aligned?

Also, What do you think about this?

**Comment by bfred-it** [comment]
I put a like on that

**Comment by HardikModha** [comment]
Raised a PR https://github.com/sindresorhus/refined-github/pull/2123

**Comment by notlmn** [comment]
If I'm not wrong, I think at some point [`parse-backticks`](https://github.com/sindresorhus/refined-github/blob/master/source/features/parse-backticks.tsx) feature can be merged into this, renaming this to `format-titles`, for formatting titles across discussion lists, issues/PRs, conflicts page, and possibly other pages, all in a single place.

**Pull request merged**